src/rgw/driver/rados/rgw_rados.cc

```diff
@@ -6070,6 +6070,31 @@ int RGWRados::Object::Read::prepare(optional_yield y, const DoutPrefixProvider *
 
   const RGWBucketInfo& bucket_info = source->get_bucket_info();
 
+  if (params.part_num) {
+    // use the manifest to redirect to the requested part number
+    if (!manifest) {
+      return -ERR_INVALID_PART;
+    }
+    RGWObjManifest::obj_iterator end = manifest->obj_end(dpp);
+    if (end.get_cur_part_id() == 0) { // not multipart
+      return -ERR_INVALID_PART;
+    }
+    if (params.parts_count) {
+      *params.parts_count = end.get_cur_part_id() - 1;
+    }
+    RGWObjManifest::obj_iterator iter = manifest->obj_find_part(dpp, *params.part_num);
+    if (iter == end) {
+      return -ERR_INVALID_PART;
+    }
+    // TODO: call get_obj_state() for the part's head object
```

Comment:
implementing `RadosReadOp` in terms of the part's head object is looking complicated because all of RGWRados' internal APIs are based on `sal::Object` - even the lowest-level ones like `get_obj_state_impl()`. this is below the line, so i don't think we should be using zipper handles at all down here, not even the `sal::Rados*` subclasses

Reply:
Yeah, early on, before there was a well defined line (or at least before I understood it), the APIs bled way down into RadosStore.  I never went back in and undid it, because it was a lot of work.

Reply:
On the plus side, you're guaranteed at this point to have RadosStore classes, so you can cheat if you need to.